### PR TITLE
make rustdoc test follow the jobserver limit of threads

### PR DESCRIPTION
**Pull request by andjo403**
fix that to many threads is executing at the same time
when rustdoc test is executed.

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
👍

File: src/librustc/session/mod.rs [hunk 1]
```diff
@@ -1128,14 +1128,21 @@ pub fn build_session_(
...
         // per-process.
-        jobserver_from_env: unsafe {
-            static mut GLOBAL_JOBSERVER: *mut Option<Client> = 0 as *mut _;
+        jobserver: unsafe {
+            static mut GLOBAL_JOBSERVER: *mut Client = 0 as *mut _;
             static INIT: std::sync::Once = std::sync::ONCE_INIT;
             INIT.call_once(|| {
-                GLOBAL_JOBSERVER = Box::into_raw(Box::new(Client::from_env()));
+                let client = Client::from_env().unwrap_or_else(|| {
+                    Client::new(::num_cpus::get()).expect("failed to create jobserver")
```
Comment (michaelwoerister) [inline src/librustc/session/mod.rs]: @andjo403, given @alexcrichton's comment below, could you change this 32? Most of the time there'll be a jobserver around anyway.

Comment (andjo403) [reply src/librustc/session/mod.rs]: @michaelwoerister have updated now and added a comment about why there is a fixed value.
Thanks for the feedback

**Comment by michaelwoerister** [comment]
Is this finished? If so, how do the changes relate to rustdoc?

**Comment by andjo403** [comment]
yes this was all that I think is needed, it works when I tested.
The problem was that when running "rustdoc test" the default is that number of cpus threads will start to compile the examples from the comments and due to there was no jobserver defined when it came to the start_async_translation it created one new jobserver for every test thread resulting in "number of cpu"*32 maximum threads created. And with this solution there is only one jobserver created due to the global variable for all compilations.
if you have a chrome browser you can see in this page https://andjo403.github.io/rs_tracing/rustdoc32.html how almost 300 threads is executing codegen on my 32 threads cpu

**Comment by michaelwoerister** [comment]
Oh ok, that makes sense! Great find!

@bors r+

**Comment by alexcrichton** [comment]
@bors: retry

* crates.io was down

**Pull request merged**